mysite/code/services/StitchDataSender.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+use Sminnee\StitchData\StitchApi;
+
+/**
+ * Sends package data to StitchData
+ */
+class StitchDataSender
+{
+
+    private $accessToken = null;
+    private $clientID = null;
+    private $tableName = null;
+
+    public function __construct()
+    {
+        // Use environment defines for client ID and access token by default
+        if (defined('STITCHDATA_CLIENT_ID')) {
+            $this->clientID = STITCHDATA_CLIENT_ID;
+        }
+        if (defined('STITCHDATA_ACCESS_TOKEN')) {
+            $this->accessToken = STITCHDATA_ACCESS_TOKEN;
+        }
+    }
+
+    /**
+     * Set the client ID of your StitchData account.
+     * @return $this, for use with fluent syntax
+     */
+    public function setClientID($clientID)
+    {
+        $this->clientID = $clientID;
+        return $this;
+    }
+
+    /**
+     * Set the access token for the StitchData Import API
+     * @return $this, for use with fluent syntax
+     */
+    public function setAccessToken($accessToken)
+    {
+        $this->accessToken = $accessToken;
+        return $this;
+    }
+
+    /**
+     * Set the table name to write packages to
+     * @return $this, for use with fluent syntax
+     */
+    public function setTableName($tableName)
+    {
+        $this->tableName = $tableName;
+        return $this;
+    }
+
+    /**
+     * Get the client ID of your StitchData account.
+     * @return string
+     */
+    public function getClientID()
+    {
+        return $this->clientID;
+    }
+
+    /**
+     * Get the access token for the StitchData Import API
+     * @return string
+     */
+    public function getAccessToken()
+    {
+        return $this->accessToken;
+    }
+
+    /**
+     * Get the table name to write packages to
+     * @return string
+     */
+    public function getTableName()
+    {
+        return $this->tableName;
+    }
+
+    /**
+     * Send the given package to the StitchData API
+     */
+    public function sendAddon(Addon $package)
+    {
+        // If unconfigured, silently no-op
+        if (!$this->clientID || !$this->accessToken || !$this->tableName) {
+            return;
+        }
+
+        $stitch = new StitchApi($this->clientID, $this->accessToken);
```

Comment:
Considering addons are usually built in bulk wouldn't in be better to instantiate this once rather than per addon send?

Reply:
The StitchDataSender is already fetched as a singleton, so I've made a change to retrieve this as an instance property `getClient()` instead